src/UIComponents/SelectDataset.jsx

```diff
@@ -51,6 +54,9 @@ class SelectDataset extends Component {
       download: true
     });
 
+    // We reset state, but we want to keep hiding the trainer selection.
+    this.props.setSelectedTrainer("knnClassify");
```

Comment:
How do we know what to see this to?

Reply:
Good point.  I'll change technique to use what was specified by the optional `mode`.